### PR TITLE
Add an endpoint to search by accommodation merge history

### PR DESCRIPTION
**Pull request by oleg-n-k**
Add an endpoint to search by accommodation merge history

### CODE REVIEW

**Comment by oleg-n-k** [comment]
The response model will be added to the mapper contracts after the code review

File: HappyTravel.Nakijin.Api/Controllers/AdministratorControllers/AccommodationMergeHistoryController.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+using HappyTravel.Nakijin.Api.Services.AdministratorServices.Management.Accommodations;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace HappyTravel.Nakijin.Api.Controllers.AdministratorControllers;
+
+[ApiController]
+[ApiVersion("1.0")]
+[Route("api/{v:apiVersion}/admin/accommodations/merge/history")]
```
Comment (hasmikharutyunyan) [inline HappyTravel.Nakijin.Api/Controllers/AdministratorControllers/AccommodationMergeHistoryController.cs:13]: maybe `merge-history`  ?

File: HappyTravel.Nakijin.Api/Models/AccommodationsMerge/AccommodationMergeHistoryAccommodationData.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    public string Country { get; init; }
+
+    public string Address { get; init; }
+
+    public AccommodationRatings Rating { get; init; }
+
+    public HappyTravel.Geography.GeoPoint Coordinates { get; init; }
+
+    public Dictionary<string, string> SupplierCodes { get; init; }
```
Comment (hasmikharutyunyan) [inline HappyTravel.Nakijin.Api/Models/AccommodationsMerge/AccommodationMergeHistoryAccommodationData.cs:27]: `SupplierAccommodationCodes`  ?

File: HappyTravel.Nakijin.Api/Services/AdministratorServices/Management/Accommodations/AccommodationMergeHistorySearchQueryableBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        => queryable
+            .FilterByName(accommodationNameQuery)
+            .FilterByCountries(countryCodes)
+            .FilterBySuppliers(supplierCodes)
+            .FilterByDate(fromDate, toDate);
+    
+    
+    private static IQueryable<AccommodationMergeHistoryEvent> FilterByName(this IQueryable<AccommodationMergeHistoryEvent> searchQuery, string? accommodationNameQuery) 
+        => !string.IsNullOrWhiteSpace(accommodationNameQuery) 
+            ? searchQuery.Where(h => EF.Functions.ILike(h.SourceAccommodation.KeyData.DefaultName, $"{accommodationNameQuery}%"))
```
Comment (hasmikharutyunyan) [inline HappyTravel.Nakijin.Api/Services/AdministratorServices/Management/Accommodations/AccommodationMergeHistorySearchQueryableBuilder.cs:26]: maybe search in merged accommodations also ?

Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Api/Services/AdministratorServices/Management/Accommodations/AccommodationMergeHistorySearchQueryableBuilder.cs:26]: discussed

**Review by hasmikharutyunyan — APPROVED** [review]

**Pull request merged**